NuKeeper.Update/Process/MonoExecutor.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuKeeper.Abstractions.Logging;
+using NuKeeper.Update.ProcessRunner;
+
+namespace NuKeeper.Update.Process
+{
+    public class MonoExecutor : IMonoExecutor
+    {
+        private readonly INuKeeperLogger _logger;
+        private readonly IExternalProcess _externalProcess;
+
+        private readonly AsyncLazy<bool> _checkMono;
+
+        public MonoExecutor(INuKeeperLogger logger, IExternalProcess externalProcess)
+        {
+            _logger = logger;
+            _externalProcess = externalProcess;
+            _checkMono = new AsyncLazy<bool>(CheckMonoExists);
+        }
+
+        public async Task<bool> CanRun()
+        {
+            return await _checkMono;
+        }
+
+        public async Task<ProcessOutput> Run(string workingDirectory, string command, string arguments, bool ensureSuccess)
+        {
+            _logger.Normal($"Using Mono to run '{command}'");
+
+            if (!await CanRun())
+            {
+                _logger.Error($"Cannot run '{command}' on Mono since Mono installation was not found");
+                throw new InvalidOperationException("Mono installation was not found");
+            }
+
+            return await _externalProcess.Run(workingDirectory, "mono", $"{command} {arguments}", ensureSuccess);
+        }
+
+        private async Task<bool> CheckMonoExists()
+        {
+            var result = await _externalProcess.Run("", "mono", "--version", false);
```

Comment:
Wouldn't it be nicer to create a dedicated command for this specific part? See NuKeeper.Update.Process folder.

Reply:
Gave it a bit of thought. I can introduced `CheckMonoExistanceCommand` or something similarly named, but no sure if it makes it nicer. 
What I think is nice is that the whole mono thing is very much self contained and if we need to change, for example, how it's being discovered (i.e. allow to pass an argument that contains the path to it) it's just all here.

Let me know what do you think.